src/routes/+page.svelte

```diff
@@ -304,11 +307,20 @@
             textCopied = false;
         }, 3000);
     }
+
+    function handleBackNavigation(event) {
+        event.preventDefault();
+        if ($contentsStack.length > 0) {
+            const menuId = contentsStack.popItem();
+            goto(`${base}/contents/${menuId}`);
+        }
+    }
+    $: showBackButton = contents?.features?.['navigation-type'] === 'up' && $contentsStack.length > 0
```

Comment:
I suspect this is where the change would need to be made to make sure the back button always shows if the navigation type is up.  Then changes made so that if it is pressed and the stack is empty, it goes to the first contents screen.